code/obj/item/clothing/chameleon_js.dm

```diff
@@ -269,3 +272,1659 @@
+		usr.set_clothing_icon_dirty()
+
+/datum/chameleon_belt_pattern
+	var/name = "utility belt"
+	var/desc = "Can hold various small objects."
+	var/icon_state = "utilitybelt"
+	var/item_state = "utility"
+	var/sprite_item = 'icons/obj/items/belts.dmi'
+	var/sprite_worn = 'icons/mob/clothing/belt.dmi'
+	var/sprite_hand = 'icons/mob/inhand/hand_storage.dmi'
+
+	ceshielded
+		name = "aurora MKII utility belt"
+		desc = "An utility belt for usage in high-risk salvage operations. Contains a personal shield generator. Can be activated to overcharge the shields temporarily."
+		icon_state = "cebelt"
+		item_state = "cebelt"
+
+	security
+		name = "security toolbelt"
+		desc = "For the trend-setting officer on the go. Has a place on it to clip a baton and a holster for a small gun."
+		icon_state = "secbelt"
+		item_state = "secbelt"
+
+	medical
+		name = "medical belt"
+		desc = "A specialized belt for treating patients outside medbay in the field. A unique attachment point lets you carry defibrillators."
+		icon_state = "injectorbelt"
+		item_state = "medical"
+
+	shoulder_holster
+		name = "shoulder holster"
+		icon_state = "shoulder_holster"
+		item_state = "shoulder_holster"
+
+	miner
+		name = "miner's belt"
+		desc = "Can hold various mining tools."
+		icon_state = "minerbelt"
+		item_state = "mining"
+
+/obj/item/storage/backpack/chameleon
+	name = "backpack"
+	desc = "A thick, wearable container made of synthetic fibers, able to carry a number of objects comfortably on a crewmember's back."
+	icon_state = "backpack"
+	item_state = "backpack"
+	inhand_image_icon = 'icons/mob/inhand/hand_storage.dmi'
+	wear_image_icon = 'icons/mob/clothing/back.dmi'
+	uses_multiple_icon_states = 1
+	var/list/clothing_choices = list()
+	spawn_contents = list()
+
+	New()
+		..()
+		var/obj/item/remote/chameleon/remote = new /obj/item/remote/chameleon(src.loc)
+		SPAWN_DBG(0)
+			remote.connected_backpack = src
+
+		var/obj/item/clothing/under/chameleon/jumpsuit = new /obj/item/clothing/under/chameleon(src)
+			remote.connected_jumpsuit = jumpsuit
+
+		var/obj/item/clothing/head/chameleon/hat = new /obj/item/clothing/head/chameleon(src)
+			remote.connected_hat = hat
+
+		var/obj/item/clothing/suit/chameleon/suit = new /obj/item/clothing/suit/chameleon(src)
+			remote.connected_suit = suit
+
+		var/obj/item/clothing/glasses/chameleon/glasses = new /obj/item/clothing/glasses/chameleon(src)
+			remote.connected_glasses = glasses
+
+		var/obj/item/clothing/shoes/chameleon/shoes = new /obj/item/clothing/shoes/chameleon(src)
+			remote.connected_shoes = shoes
+
+		var/obj/item/storage/belt/chameleon/belt = new /obj/item/storage/belt/chameleon(src)
+			remote.connected_belt = belt
+
+		var/obj/item/clothing/gloves/chameleon/gloves = new /obj/item/clothing/gloves/chameleon(src)
+			remote.connected_gloves = gloves
+
+		for(var/U in (typesof(/datum/chameleon_backpack_pattern)))
+			var/datum/chameleon_backpack_pattern/P = new U
+			src.clothing_choices += P
+		return
```

Comment:
Few things about these lines:
- I tested this without the SPAWN and it seemed to work fine, so I'd remove that unless you have a specific reason for it
- Indents aren't needed
- empty return at the end of a proc doesn't do anything
```suggestion
		remote.connected_backpack = src

		var/obj/item/clothing/under/chameleon/jumpsuit = new /obj/item/clothing/under/chameleon(src)
		remote.connected_jumpsuit = jumpsuit

		var/obj/item/clothing/head/chameleon/hat = new /obj/item/clothing/head/chameleon(src)
		remote.connected_hat = hat

		var/obj/item/clothing/suit/chameleon/suit = new /obj/item/clothing/suit/chameleon(src)
		remote.connected_suit = suit

		var/obj/item/clothing/glasses/chameleon/glasses = new /obj/item/clothing/glasses/chameleon(src)
		remote.connected_glasses = glasses

		var/obj/item/clothing/shoes/chameleon/shoes = new /obj/item/clothing/shoes/chameleon(src)
		remote.connected_shoes = shoes

		var/obj/item/storage/belt/chameleon/belt = new /obj/item/storage/belt/chameleon(src)
		remote.connected_belt = belt

		var/obj/item/clothing/gloves/chameleon/gloves = new /obj/item/clothing/gloves/chameleon(src)
		remote.connected_gloves = gloves

		for(var/U in (typesof(/datum/chameleon_backpack_pattern)))
			var/datum/chameleon_backpack_pattern/P = new U
			src.clothing_choices += P
```